### PR TITLE
Sets up models for gems and engines to be loaded. Updates incinerator rake instructions. Adds ignored models.

### LINKED ISSUE #50: Fix org_incinerator to pick models from gems and engines

### PR DESCRIPTION
**Pull request by shreyansh-g24**
FIXES #50

### CODE REVIEW

File: lib/org_incineration/missed_models_finder_service.rb [hunk 1]
```diff
@@ -11,5 +11,35 @@ def initialize
...
+        NeetoPaymentsEngine::Stripe::Webhook
+        NeetoPaymentsEngine::Stripe::Refund
+        NeetoTeamMembersEngine::Permission
+        OrgIncineration::IncinerationRecord
+        Neeto::Area51::Link
+        Questions::Tag
+        NeetoTags::Tagging
+        NeetoTags::Tag
+      ]
+    end
```
Comment (shreyansh-g24) [inline lib/org_incineration/missed_models_finder_service.rb:43]: @unnitallman We can add ignored models like this. For the first group of models related to `Rails`, we'd need to hardcode them here.

For the second group of models, that are based on our `neeto-*-engines`, we can either:
  1. Hardcode them and manage them here as and when new models are added, OR
  2. Manage them dynamically at engine level by adding a config for each `neeto-*-engine`
    - If we follow `2.`, we can load the configs of all Engines in `org_incineration` to ignore the cumulative models mentioned in all Engines, like this,
```ruby
org_incineration_models = OrgIncineration::IncinerableConcern.ignored_models
neeto_tags_models = NeetoTags::IncinerableConcern.ignored_models
...
```

Please let me know what you think.

**Comment by akhil-gautam** [comment]
Is there any update on this? @shreyansh-g24 

CC: @unnitallman

File: lib/org_incineration/missed_models_finder_service.rb (hunk 1 above)
Comment (shreyansh-g24) [reply lib/org_incineration/missed_models_finder_service.rb:43]: As discussed on the call, adding configuration for Engines sounds good

**Comment by shreyansh-g24** [comment]
NeetoEngines
 - [x] OrgIncineration - https://github.com/bigbinary/org_incineration/pull/51
 - [ ] NeetoTags
 - [x] NeetoArea51 - https://github.com/bigbinary/neeto-area51/pull/137
 - [ ] NeetoPaymentsEngine
 - [ ] NeetoTeamMembersEngine

**Comment by shreyansh-g24** [comment]
@unnitallman Please check the PR. All the engines' incinerable_concerns are setup

**Comment by akhil-gautam** [comment]
@shreyansh-g24 https://github.com/bigbinary/neeto-team-members-engine/issues/112#issuecomment-1163981984 Is this handled?

**Comment by shreyansh-g24** [comment]
@akhil-gautam No, that issue is not handled. The result will be, if an `Organization` is incinerated, the join_table `organization_role_permissions` will contain irrelevant records that will reference deleted `OrganizationRole` records.

**Comment by shreyansh-g24** [comment]
@shreyansh-g24 _a Assigning back for testing

**Comment by shreyansh-g24** [comment]
- [ ] Ideally we should have one incineration test in all host applications, possibly in `organization_test.rb` since `incinerate!` is defined on `Organization` model. Currently this test is inconsistently present across different `neeto-apps`.

---

- [ ] If a host application includes an `Engine`, the engine should be setup properly with all the tables.

---

- [ ] `ServerSideSession` config needs to be added, either to all applications, or a specific common gem/engine,
```ruby
   "ServerSideSession": {
      joins: { user: :organization },
      where: ["users.organization_id = ?", org_id]
    },
```
---
- [ ] `NeetoTags`
  - Fix `incinerable_concern` => `NeetoTags::Tagging`
---
- [ ] `NeetoPaymentsEngine` - If the host application has payments enabled, i.e. `NeetoPaymentsEngine` tables are present, then the configuration for these models need to be there:
```ruby
"NeetoPaymentsEngine::Stripe::Transaction"
"NeetoPaymentsEngine::Stripe::Refund"
"NeetoPaymentsEngine::Stripe::Charge"
"NeetoPaymentsEngine::Stripe::Account"
```

**Comment by shreyansh-g24** [comment]
@unnitallman _a Please check and let me know your thoughts.

**Comment by akhil-gautam** [comment]
@shreyansh-g24 @unnitallman Do we have any update on this?

**Comment by shreyansh-g24** [comment]
@akhil-gautam This needs `incineration-tests` in `neeto-chat` and `neeto-invoice` before roll out. I'll try to coordinate and push it tomorrow.

**Comment by unnitallman** [comment]
> @unnitallman _a Please check and let me know your thoughts.

Looks good to me.

**Comment by shreyansh-g24** [comment]
@shreyansh-g24 _a

**Comment by shreyansh-g24** [comment]
1. [ ] neetoChat - https://github.com/bigbinary/neeto-chat-web/issues/4151
2. [ ] neetoDesk - https://github.com/bigbinary/neeto-desk-web/issues/new
3. [ ] neetoKB - https://github.com/bigbinary/neeto-kb-web/issues/new
4. [ ] neetoForm - https://github.com/bigbinary/neeto-form-web/issues/new
5. [ ] neetoInvoice - https://github.com/bigbinary/neeto-invoice-web/issues/1641
6. [ ] neetoPlanner - https://github.com/bigbinary/neeto-planner-web/issues/new
7. [ ] neetoQuiz - https://github.com/bigbinary/neeto-quiz-web/issues/new
8. [ ] neetoChangelog - https://github.com/bigbinary/neeto-changelog-web/issues/new
9. [ ] neetoInvisible - https://github.com/bigbinary/neeto-invisible-web/issues/new
10. [ ] neetoAnalytics - https://github.com/bigbinary/neeto-analytics-web/issues/new
11. [ ] neetoWireframe - https://github.com/bigbinary/neeto-wireframe-web/issues/new
12. [ ] neetoRunner - https://github.com/bigbinary/neeto-runner-web/issues/new
13. [ ] neetoPopups - https://github.com/bigbinary/neeto-popups-web/issues/new
14. [ ] neetoEngage - https://github.com/bigbinary/neeto-engage-web/issues/new
15. [ ] neetoGrow. - https://github.com/bigbinary/neeto-grow-web/issues/new
16. [ ] neetoReplay - https://github.com/bigbinary/neeto-replay-web/issues/1268
17. [ ] neetoCourse - https://github.com/bigbinary/neeto-course-web/issues/new
18. [ ] neetoCRM - https://github.com/bigbinary/neeto-crm-web/issues/new
19. [ ] neetoStore - https://github.com/bigbinary/neeto-store-web/issues/new
20. [ ] neetoCal - https://github.com/bigbinary/neeto-cal-web/issues/new
21. [ ] neetoAnalyzer - https://github.com/bigbinary/neeto-analyzer-web/issues/new
22. [ ] neetoTestify - https://github.com/bigbinary/neeto-testify-web/issues/new
23. [ ] neetoAuth - https://github.com/bigbinary/neeto-auth-web/issues/new

**Comment by shreyansh-g24** [comment]
Updates
  - `neeto-rules-engine` - https://github.com/bigbinary/neeto-rules-engine/issues/41

**Pull request merged**